crates/polars-lazy/src/frame/mod.rs

```diff
@@ -656,15 +656,44 @@ impl LazyFrame {
         Ok(())
     }
 
+    /// Stream a query result into a parquet file on an ObjectStore-compatible cloud service. This is useful if the final result doesn't fit
+    /// into memory, and where you do not want to write to a local file but to a location in the cloud.
+    /// This method will return an error if the query cannot be completely done in a
+    /// streaming fashion.
+    #[cfg(all(feature = "cloud_write", feature = "parquet"))]
+    pub fn sink_parquet_cloud(
+        mut self,
+        uri: String,
+        cloud_options: Option<polars_core::cloud::CloudOptions>,
```

Comment:
Besides the `cloud_options` parameter, this looks exactly the same as the `sink_parquet` method. Isn't there a possibility that you can just pass in the uri and the write/sink methods can determine if it's an uri or a local path?

That would make it much more easy to maintain the code, and open up all methods to natively read & write to local/cloud based sources/targets.

Reply:
While possible, there are some drawbacks of doing so:
- It is a backwards-incompatible change. Polars is still pre-1.0 but nonetheless breaking existing code is not something to do lightly.
- It makes `sink_parquet` more difficult to use for anyone _not_ using the cloud feature, since for writing to a normal file you'd now need to pass in an extra `None` every time.
- It complicates how to deal with feature flags. Currently, the feature flag `cloud_write` enables/disables `sink_parquet_cloud`. Attempting to use it without the feature flag enabled results in a compile-time error. If it would alter the internals of `sink_parquet`, the error would only be caught at runtime.


That is why I decided to keep it separate in this PR. Unifying the APIs is definitely interesting and maybe this refactor might be worth considering in a future version, but I did not want to rush this for these reasons.

Reply:
Does this PR enable us to write LazyFrames to cloud storage using sink_parquet? If so, how do I so this? I can't figure it out.